File: server/src/main/java/au/com/codeka/warworlds/server/admin/handlers/DebugSimulationQueueHandler.java
package au.com.codeka.warworlds.server.admin.handlers;

import java.util.Arrays;
import java.util.Collection;
import java.util.TreeMap;

import au.com.codeka.warworlds.common.proto.AdminRole;
import au.com.codeka.warworlds.server.handlers.RequestException;
import au.com.codeka.warworlds.server.store.DataStore;

public class DebugSimulationQueueHandler extends AdminHandler {
  /** Any role can visit this page. */
  @Override
  protected Collection<AdminRole> getRequiredRoles() {
    return Arrays.asList(AdminRole.values());
  }

  @Override
  protected void get() throws RequestException {
    TreeMap<String, Object> data = new TreeMap<>();
    data.put("stars", DataStore.i.stars().fetchSimulationQueue(100));
    render("debug/simulation-queue.html", data);
  }
}

File: server/src/main/kotlin/au/com/codeka/warworlds/server/admin/handlers/DebugSimulationQueueHandler.kt
package au.com.codeka.warworlds.server.admin.handlers

import au.com.codeka.warworlds.common.proto.AdminRole
import au.com.codeka.warworlds.server.handlers.RequestException
import au.com.codeka.warworlds.server.store.DataStore
import java.util.*

class DebugSimulationQueueHandler : AdminHandler() {
  /** Any role can visit this page.  */
  override val requiredRoles: Collection<AdminRole>?
    get() = listOf(*AdminRole.values())

  @Throws(RequestException::class)
  override fun get() {
    val data = TreeMap<String, Any>()
    data["stars"] = DataStore.i.stars().fetchSimulationQueue(100)
    render("debug/simulation-queue.html", data)
  }
}